backend/src/test/java/codezap/template/service/TemplateServiceTest.java

```diff
@@ -7,272 +7,566 @@
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+
+            assertAll(
+                    () -> assertThat(actual.getContent()).hasSize(2),
+                    () -> assertThat(actual.getContent())
+                            .allMatch(template -> template.getMember().getId().equals(member1.getId()))
+            );
+        }
+
+        @Test
+        @DisplayName("검색 기능: 키워드로 템플릿 조회 성공")
+        void findAllSuccessByKeyword() {
+            saveInitialData();
+            Long memberId = null;
+            String keyword = "Template";
+            Long categoryId = null;
+            List<Long> tagIds = null;
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+            for (Template template : actual.getContent()) {
+                System.out.println(template.getTitle());
+            }
+
+            assertAll(
+                    () -> assertThat(actual.getContent())
+                            .allMatch(template ->
+                                    template.getTitle().contains(keyword) || template.getDescription()
+                                            .contains(keyword)),
+                    () -> assertThat(actual.getContent()).hasSize(3)
+            );
+        }
+
+        @Test
+        @DisplayName("검색 기능: 카테고리 ID로 템플릿 조회 성공")
+        void findAllSuccessByCategoryId() {
+            saveInitialData();
+            Long memberId = null;
+            String keyword = null;
+            Long categoryId = category1.getId();
+            List<Long> tagIds = null;
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+
+            assertAll(
+                    () -> assertThat(actual.getContent()).hasSize(2),
+                    () -> assertThat(actual.getContent())
+                            .allMatch(template -> template.getCategory().getId().equals(category1.getId()))
+            );
+        }
+
+        @Test
+        @DisplayName("검색 기능: 태그 ID 목록으로 템플릿 조회, 모든 태그를 가진 템플릿만 조회 성공")
+        void findAllSuccessByTagIds() {
+            saveInitialData();
+            Long memberId = null;
+            String keyword = null;
+            Long categoryId = null;
+            List<Long> tagIds = List.of(tag1.getId(), tag2.getId());
+            Pageable pageable = PageRequest.of(0, 10);
+
+            Page<Template> actual = sut.findAll(memberId, keyword, categoryId, tagIds, pageable);
+
+            assertAll(
+                    () -> assertThat(actual.getContent())
+                            .containsExactlyInAnyOrder(
+                                    templateRepository.fetchById(1L),
+                                    templateRepository.fetchById(2L)),
```

Comment:
이것도 취향 차이지만...
한줄로 적어도 좋을 것 같아요!
위의 allMatch보다 길이가 짧아서 가능할 것 같은...??

```suggestion
                            .containsExactlyInAnyOrder(templateRepository.fetchById(1L),templateRepository.fetchById(2L)),
```

Reply:
이거...한 줄로 쓰면 자동으로 줄바꿈 됩니다..😭